index.js

```diff
@@ -1,18 +1,131 @@
+let maxTries = (Math.round((Math.log2((max)+1))));
+
+function getRandomIntInclusive(min, max) {
+  min = Math.ceil(min);
+  max = Math.floor(max);
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     rl.question(questionText, resolve);
   });
 }
 
 start();
-
+  
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
+  console.log("Let's play a game where you (human user) make up a number and I (computer program) try to guess it.")
+  console.log("Please think of a number between " + min + " and " + max + " (inclusive), in this " + range + " number range.");
+  let secretNumber = await ask("What is your secret number?\n(This Javascript (JS) program (pgm) won't peek...)\n");
+  console.log('FYI, you entered: ' + secretNumber + ' (Again, this will be kept secret until the end!)'); 
+
+if ((secretNumber >= min) && (secretNumber <= max)) {   
+
+// mainGuessing 
+  cpuGuessNum = getRandomIntInclusive(min, max);   // Guessing a random number in the range to start
+  console.log('(This program currently randomly guesses only the first try) Is it... ' + cpuGuessNum); 
+  let pgmTries = 1; 
+  let userRespYorN = await ask('? (Please type a Y or a N only) -> ');
+    let userRespYorNUC = userRespYorN.toUpperCase().trim();  // to capitalize letters and also trim spaces
+    
+  if (userRespYorNUC === 'Y') {
+    console.log('You stated that this was the correct number, Yea!');  
+    console.log('The program has now tried: ' + pgmTries + ' times.'); // game exits since it's done
+    process.exit();
+  } else if (userRespYorNUC === 'N') {
+    console.log('You stated it was NOT correct, or ' + userRespYorNUC + ' , Sorry.');
+    // keepGuessing
+
+  pgmTries = pgmTries + 1;  // add another try
+  console.log('The program will try another time - This time makes: ' + pgmTries + ' times.');
+  console.log('The maximum number of tries this program should theoretically take is only: ' + maxTries);
+
+  // it was a N, then continue - 
+  if (userRespYorNUC === 'N') { 
+    continueTrying = (true);}    
+  let userRespHorL = await ask("Is it higher (H) or lower (L)? ");
+    
+  // in case the user used lower case - 
+  let userRespHorLUC = userRespHorL.toUpperCase();  
+    
+  // goes to High/Low Logic -  
+    if (userRespHorLUC == 'H') {  // goes to Higher logic 
+   (gotoHighLoop = (true)); 
+     // get a Higher range --
+     let newMin = (cpuGuessNum + 1); // causes it to not include cur low guess
+     let newRange = (max - newMin);
+     cpuGuessNum = (newRange/2);
+  } else if (userRespHorLUC == 'L') {  // goes to Lower Logic
+    (gotoLowLoop = (true));
```

Comment:
You're defining "gotoHighLoop" & "gotoLowLoop", but never using them. This is where you start running into some problems.